rlclientlib/logger/async_batcher.h

```diff
@@ -77,9 +78,13 @@ namespace reinforcement_learning { namespace logger {
     queue_mode_enum _queue_mode;
     std::condition_variable _cv;
     std::mutex _m;
+    std::mutex _counter_mutex;
     utility::object_pool<utility::data_buffer> _buffer_pool;
     const char* _batch_content_encoding;
     float _subsample_rate;
+    events_counter_status _events_counter_status;
+    uint64_t _event_index = 0;
```

Comment:
Do you really need uint64_t? Can unit32_t be enough?

Reply:
As per this conversation, uint64_t seems more reasonable

Reply:
50 RPS is quite realistic.
It will give us 50 * 3600 * 24 ~ 4 millions
uint32 limit is 4 billions which means 3 years of traffic with 50 RPS (or less with higher traffic).
it seems like it should be enough in most of the cases but quite close to realistic limits, so maybe let's just do it safe?

Reply:
I missed your previous comments. I am fine with the justification.